CONTRIBUTING.md

```diff
@@ -1,9 +1,203 @@
+## Issues and bug reporting
+Please report all bugs using the Issues tab on the GitHub web-page. It will help us to collaborate on issues more promptly. One of our gate-keepers (developers responsible for quality and the repository) will review the issue and assign a label such as _bug_, _enhancement_, _help wanted_ or _wontfix_.
+# How to contribute
+You can either file a bug, help fix a bug or propose a new feature (or enhancement) and implement it yourself. If you want to contribute please:
+* Bug reports: File a bug report in the Issues tab of this repo to let us know there are problems with our code.
+  * Make sure your bug report contains a simple description of the problem.
+  * Include information about the host computer's configuration and OS or VM used.
+  * Include information about the application's version. All applications should have at least a ``--version`` switch you can use to check the version.
+  * Copy/paste useful console dumps and configuration files' content. Please use [fenced code blocks](https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks) to encapsulate code snippets.
+* New features or bug fix: Create a pull request with your changes.
+* General feedback: Give your feedback by posting your comments on existing pull requests and issues.
+
+## Simple workflow for bugifx
+* Select an issue to fix from open issues.
+* Fork repository you wish to modify.
+* Clone locally your fork, create a separate branch for issue to fix:
+
+Note: In this example we will fix issue #38.
+```
+$ git clone <fork-repo-link>
+$ git checkout -b issue_38
+... add changes locally to fix an issue
+```
+
+* Add and commit your changes.
+
+```
+$ git add .
+$ git commit -m "Add fix for issue #38" -m "More verbose explanation of the change/fix"
+$ git push origin issue_38
+
+```
+
+* Push changes to GitHub.
+* Create pull request from GitHub webpage (your fork's dashboard).
+
+### Branch naming conventions
+We prefer is you use standardised naming convention when creating pull requests.
+Below few example of branch names' prefixes you could use when creating pull request from your fork:
+* ```issue_``` - branch with fix for issue. E.g. ```issue_38```.
+* ```doc_``` - documentation update. E.g. ```doc_add_faq```.
+* ```devel_``` - development of a new feature. E.g. ```devel_udp_client_test```.
+* ```test_``` - when pull request will consist of only new/updates to test cases. E.g. ```test_paralllel_execution```.
+
+## Coding style and coding rules
+This chapter attempts to explain the basic styles and patterns that are used in mbed test tools projects. he following norms should be followed for new code, and for code that needs clean-up.
```

Comment:
The following rules should be followed for producing new code and updating existing code.